harvester/mets.py

```diff
@@ -61,13 +61,59 @@ def _ensure_files(self):
         )
 
         self._files = []
-        for file_element in files:
-            self._files.append(
-                File.file_from_element(
-                    file_element,
-                    self.binding_dc_identifier,
+
+        self._add_access_image_files(file_elements=files)
+        self._add_alto_files_for_images()
+
+    def _add_access_image_files(self, file_elements):
+        """
+        Add an AccessImageFile to self._files for every access image in file_elements
+        """
+        for file_element in file_elements:
+            children = file_element.getchildren()
+            if len(children) != 1:
+                raise METSLocationParseError(
+                    "Expected 1 location, found {len(children)}"
+                )
+            location = file_element.xpath("./*/@*[local-name()='href']")[0]
+            parent = file_element.getparent()
+
+            filegrp_use = parent.attrib["USE"]
+            filegrp_id = parent.attrib["ID"]
+
+            if filegrp_use in ["Images", "reference"] and filegrp_id in [
+                "IMGGRP",
+                "ACIMGGRP",
+            ]:
+                page_number = int(file_element.attrib["SEQ"])
```

Comment:
If this raises an exception, do we ultimately get a failed task? I think so, via `_ensure_files()` -> `files()` -> `SavePageFilesSFTPOperator.execute()` -> `SaveAltosSFTPOperator.execute()` -> `StowBindingBatchOperator()`. What I mean is, are we so sure of these always existing that we don't want to check?

Reply:
Yea, the task will now fail if we can't find the page number.

NLF gave us a soft promise that all the METSes currently in the collection should have them, and at the time we don't really have a good alternative for figuring out the mapping between elements that isn't complex (mapping the file elements to other elements in other sections of the METS via their identifiers and deducing the URLs based on that) and still uncertain (we still don't know if all that metadata is consistent and always available). So even if we checked, at the moment we wouldn't have a good alternative to just manually raising an exception.

If it turns out that SEQs are still missing in the new collection, then we need to figure out what those bindings are like and how we can work with them, but that requires actually seeing what metadata they actually have.